### PR TITLE
Update api key

### LINKED ISSUE #104: Update `api_key` and validate permissions
`core` generates `ApiKey`'s before the `user_id` has been assigned in order to validate phone numbers. There're two methods we need:

- [x]  One method to update an `ApiKey`. Once the phone number has been validated, the `ApiKey` is associated to its corresponding user. So we need a method to update an `ApiKey` along with its `metadata`.

```python
import cuenca

cuenca.ApiKey.update(metadata='ANYTHING', user_id=USER_ID)
```
The corresponding permissions for a `user` to update an `api_key` would be managed by `authed`

### PR DESCRIPTION
**Pull request by ricardo8990**
Agrega métodos para actualizar un `ApiKey` y para validar los permisos de credenciales.

closes #104

### CODE REVIEW

File: cuenca/resources/api_keys.py [hunk 1]
```diff
@@ -37,5 +38,20 @@ def deactivate(cls, api_key_id: str, minutes: int = 0) -> 'ApiKey':
...
+        api_key_id: str,
+        metadata: Optional[dict] = None,
+        user_id: Optional[str] = None,
+    ) -> 'ApiKey':
+        """
+        If the current user has enough permissions, it associates an ApiKey to
+        the `user_id` or updates the correspoding metadata
+        """
+        req = ApiKeyUpdateRequest(metadata=metadata, user_id=user_id)
+        resp = cls._update(api_key_id, **req.dict(exclude_none=True))
```
Comment (matin) [inline cuenca/resources/api_keys.py:56]: what about making `exclude_none=True` the default for all request validations the way we did for queries: https://github.com/cuenca-mx/cuenca-validations/blob/main/cuenca_validations/types/queries.py#L32

**Review by matin — COMMENTED** [review]
seems reasonable. just have one question on approach for request validation.

File: cuenca/resources/api_keys.py (hunk 1 above)
Comment (felipaoo) [reply cuenca/resources/api_keys.py:56]: it make sense, I will send a PR for that in `cuenca-validations`

**Review by matin — COMMENTED** [review]
Something that I missed the guest time around: we should maintain consistency in the resource methods. We can't have some but not others support a local session.

File: cuenca/resources/api_keys.py [hunk 2]
```diff
@@ -24,8 +25,8 @@ def active(self) -> bool:
         )
 
     @classmethod
-    def create(cls) -> 'ApiKey':
-        return cast('ApiKey', cls._create())
+    def create(cls, *, session: Session = global_session) -> 'ApiKey':
+        return cast('ApiKey', cls._create(session=session))
 
     @classmethod
     def deactivate(cls, api_key_id: str, minutes: int = 0) -> 'ApiKey':
```
[the remark this answers is not in the document]
Comment (felipaoo) [reply cuenca/resources/api_keys.py:31]: It's because we are not using those method with local session right now. I will add it to have consistency

**Review by matin — APPROVED** [review]

**Review by felipaoo — APPROVED** [review]

**Pull request merged**